app/src/main/java/com/example/ithaca_transit_android_v2/views/RouteCardCompact.kt

```diff
@@ -0,0 +1,137 @@
+package com.example.ithaca_transit_android_v2.views
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.view.View
+import android.widget.*
+import androidx.annotation.NonNull
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.example.ithaca_transit_android_v2.NetworkUtils
+import com.example.ithaca_transit_android_v2.R
+import com.example.ithaca_transit_android_v2.models.Coordinate
+import com.example.ithaca_transit_android_v2.models.Route
+import com.example.ithaca_transit_android_v2.presenters.RouteCardPresenter
+import com.google.android.material.bottomsheet.BottomSheetBehavior
+import io.reactivex.disposables.Disposable
+import kotlinx.android.synthetic.main.route_card_compact.*
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.runBlocking
+import kotlin.collections.ArrayList
+
+class RouteCardCompact : AppCompatActivity() {
```

Comment:
quick q, does this class control just a singular route card or the entire bottom sliding sheet? it seems to contain code for both, but you should try to avoid that as it makes the code more disorganized

Reply:
This class handles the entire sliding sheet. The rvadapter handles the logic for each individual route card.

Reply:
ah i see, make sure to add a comment above on line 23 explaining the purpose of the class (and do the same with rvadapter)